File: basketball_stats_app/lib/models/player.dart
// lib/models/player.dart
import 'package:json_annotation/json_annotation.dart';
import 'team.dart';

part 'player.g.dart';

@JsonSerializable()
class Player {
  @JsonKey(name: '_id')
  final String? id;
  final String name;
  final String position;
  final int? number;
  final String? nationality;
  final int? height;
  final int? weight;
  final Team team; // Team object
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Player({
    this.id,
    required this.name,
    required this.position,
    this.number,
    this.nationality,
    this.height,
    this.weight,
    required this.team,
    this.createdAt,
    this.updatedAt,
  });

  factory Player.fromJson(Map<String, dynamic> json) {
    print('Player.fromJson called with: $json'); // Log per debug
    return _$PlayerFromJson(json);
  }

  Map<String, dynamic> toJson() => _$PlayerToJson(this);
}

File: basketball_backend/models/Player.js
// models/Player.js
const mongoose = require('mongoose');

const playerSchema = mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Inserisci il nome del giocatore'],
  },
  position: {
    type: String,
    required: [true, 'Inserisci la posizione del giocatore'],
  },
  number: { type: Number, required: false }, // numero di maglia
  nationality: { type: String, required: false }, // nazionalità
  height: { type: Number, required: false }, // altezza in cm
  weight: { type: Number, required: false }, // peso in kg
  team: { type: mongoose.Schema.Types.ObjectId, ref: 'Team', required: true },
  createdAt: {
    type: Date,
    default: Date.now,
  },
  updatedAt: {
    type: Date,
    default: Date.now,
  },
});

const Player = mongoose.model('Player', playerSchema); // Corretto: usa playerSchema

module.exports = Player;
